client/flutter/lib/api/content_loading.dart

```diff
@@ -105,3 +111,48 @@ class ContentLoading {
     return '${name}.$suffix.yaml';
   }
 }
+
+class WhoCacheManager extends BaseCacheManager {
+  static const key = "whoCache";
+
+  static WhoCacheManager _instance;
+
+  // Attempting to change the maxAgeCacheObject parameter here has no effect
+  // on http content, for which the manager uses the cache control max-age header.
+  WhoCacheManager._init() : super(key);
+
+  factory WhoCacheManager() {
+    if (_instance == null) {
+      _instance = WhoCacheManager._init();
+    }
+    return _instance;
+  }
+
+  // This changes the behavior of the base method to wait for an update of the
+  // expired file before returning, rather than favoring the cached version.
+  @override
+  Future<File> getSingleFile(String url, {Map<String, String> headers}) async {
+    var cacheFile = await getFileFromCache(url);
+    if (cacheFile != null) {
+      if (cacheFile.validTill.isBefore(DateTime.now())) {
+        // Wait for the refreshed file.
+        await webHelper.downloadFile(url, authHeaders: headers);
+      }
+      return cacheFile.file;
+    }
+    try {
+      print("Downloading file: $url");
+      var download = await webHelper.downloadFile(url, authHeaders: headers);
+      return download.file;
+    } catch (e) {
+      print("Error downloading file: $url, $e");
+      return null;
+    }
+  }
+
+  // Store files in the temporary system path (default behavior)
+  Future<String> getFilePath() async {
+    var directory = await getTemporaryDirectory();
```

Comment:
What are the security effects of doing this?

Reply:
This returns a private, per-app temp storage area.  On iOS this uses NSCaches with the app's bundle id to get this location.  This package (path_provider) is maintained by the official flutter developer team.  This particular behavior is identical to the default caching package impl, I was simply required to repeat it here in the subclass.

Reply:
getTemporaryDirectory [is analogous to](https://flutter.dev/docs/cookbook/persistence/reading-writing-files#1-find-the-correct-local-path):

> A temporary directory (cache) that the system can clear at any time. On iOS, this corresponds to the [NSCachesDirectory](https://developer.apple.com/documentation/foundation/nssearchpathdirectory/nscachesdirectory). On Android, this is the value that [getCacheDir()](https://developer.android.com/reference/android/content/Context#getCacheDir()) returns.

It appears that neither can be written to be other apps, although files in these directories can be deleted by the system / by the behavior of another app if there is a need to free up more storage space.